scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -549,6 +554,46 @@ def add_testsuites(self, testsuite_filter=None):
                                 suite.platform_allow,
                                 f"platform_allow in {suite.name}")
 
+                        if self.options.log_level == "QA":
+                            if len(suite.platform_allow) > 10:
+                                logger.qa(f"Too many entries in allow_platform in {suite.name}: {len(suite.platform_allow)}")
+
+                            if len(suite.integration_platforms) > 5:
+                                logger.qa(f"Too many integration platforms in {suite.name}: {len(suite.integration_platforms)}")
+
+                            if len(suite.platform_exclude) > 5:
+                                logger.qa(f"Too many excluded platforms in {suite.name}: {len(suite.platform_exclude)}")
+
+                            if len(suite.platform_allow) > 1 and not suite.integration_platforms:
+                                logger.qa(f"platform_allow in {suite.name} without integration_platforms")
+
+
+                            if suite.build_only and not "build" in suite.id:
```

Comment:
some more comments would help to understand the guidelines. Is the idea behind it that all scenarios with "build_only: true" have "build" in their name? If so, the printed output doesn't reflect this.

Reply:
that is a hack right now, there are build tests, i.e. things we only build and should NEVER run and we need to mark those differently and distinguish them from build_only set for other reasons or no reason at all. I am thinking to introduce and enforce 'harness: build` for such tests and stop using `build_only`...